storm-server/src/main/java/org/apache/storm/daemon/supervisor/Slot.java

```diff
@@ -497,7 +497,6 @@ static DynamicState cleanupCurrentContainer(DynamicState dynamicState, StaticSta
         assert(dynamicState.container.areAllProcessesDead());
         
         dynamicState.container.cleanUp();
-        staticState.localizer.releaseSlotFor(dynamicState.currentAssignment, staticState.port);
```

Comment:
cleanupCurrentConatiner gets used in many different locations, not just during the blob update.  We need to release the slot when the container is killed or reference counting will be off.

Reply:
yeah, but `staticState.localizer.releaseSlotFor` only remove topology base blobs[jar/code/conf] which are already taken care of by releaseSlotFor itself.

Reply:
That is not true.

https://github.com/danny0405/storm/blob/a4e659b5073794396ea23e3dd7b79c00536fc3fe/storm-server/src/main/java/org/apache/storm/localizer/AsyncLocalizer.java#L505-L511

The code first removes the base blobs reference counts, but then it decrements a reference count for other blobs too.